### ISSUE TITLE
[Bug] PSD annotation promotion with negative floats.

### ISSUE THREAD

**Issue by daniel-dodd** [wilson-labs/cola#24]
# 🐛 Bug PSD annotation promotion with negative floats.

Maybe more of a question about the annotation "mental model", but should the following be expected behaviour? :)

## To reproduce

A simple example demonstrating negative floats against a PSD matrix:
```python
import jax.numpy as jnp
from cola.ops import Identity

n = 123
I = Identity(shape=(n, n), dtype=jnp.float64) # has {PSD, Unitary} annotations by default.

negative_I = -1.0 * I # negating so now NSD.

print(negative_I.annotations)  # inspecting annotations.
```

```
# {PSD, Unitary}
```

But `negative_I` is not PSD.

## Expected Behavior

- Would probably expect NSD annotation or no PSD annotation to be present.

## System information

**Please complete the following information:**
- CoLA Version (run `print(cola.__version__)`: 0.1.0 / commit 56b913b82d1bf2d41b67f9748357fe626bd6b782
- JaX and/or PyTorch Version (run `print(jax.__version__)` and/or `print(torch.__version__)` N/a here.
- Computer OS: N/a here.

**Comment by mfinzi** [comment]
Yeah that's a good point I will need to investigate. Previously I had checking on the values to propagate the types but this causes a problem with Jax autodiff/compilation.

Probably we should just invalidate the annotations on multiplication with scalarmul but need to decide how to handle this situation more generally. Actually, I wonder if Jax.lax.cond would be applicable here

**Comment by mfinzi** [comment]
I suppose [jax.lax.cond](https://jax.readthedocs.io/en/latest/_autosummary/jax.lax.cond.html) will not work here since the output (the set of annotations) is not a pytree. If the annotation [objects](https://github.com/wilson-labs/cola/blob/main/cola/annotations.py) were trivial (empty) pytrees with different types, I think this would still not work because the two paths of cond need to have identical output types right?
